driver/mounter.go

```diff
@@ -194,9 +202,28 @@ func (m *mounter) IsMounted(source, target string) (bool, error) {
 			err, findmntCmd, string(out))
 	}
 
-	if strings.TrimSpace(string(out)) == "" {
-		return false, nil
+	var resp *findmntResponse
+	err = json.Unmarshal(out, &resp)
+	if err != nil {
+		panic(err)
 	}
 
-	return true, nil
+	targetFound := false
+	for _, fs := range resp.FileSystems {
+		// check if the mount is propagated correctly. It should be set to shared.
+		if fs.Propagation != "shared" {
+			return true, fmt.Errorf("mount propagation for target %q is not enabled", target)
+		}
+
+		// the mountpoint should match as well
+		if fs.Target == target {
+			targetFound = true
```

Comment:
`return true, nil` and get rid of `targetFound`

Reply:
Good call, but we can't do it because there might be two mount points. If the first one is `shared` and it fits the target, if we early return in this case, we might miss the second one (which could be `private` for example). However I removed this piece:

```
	if targetFound {
		return true, nil
	}

	return false, nil
```

to 

```
	return targetFound, nil
```

Reply:
❤️